src/LayoutManager/LayoutManager.ts

```diff
@@ -0,0 +1,294 @@
+      context,
+      target.getObjects()
+    );
+    if (!result) {
+      return;
+    }
+    const nextCenter = new Point(result.centerX, result.centerY);
+    const correction = new Point(
+      result.correctionX ?? 0,
+      result.correctionY ?? 0
+    );
+    const relativeCorrection = new Point(
+      result.relativeCorrectionX ?? 0,
+      result.relativeCorrectionY ?? 0
+    );
+    const offset =
+      context.type === 'initialization' && context.objectsRelativeToGroup
+        ? new Point()
+        : prevCenter
+            .subtract(nextCenter)
+            .add(correction)
+            .transform(
+              // in `initialization` we do not account for target's transformation matrix
+              context.type === 'initialization'
+                ? iMatrix
+                : invertTransform(target.calcOwnMatrix()),
+              true
+            )
+            .add(relativeCorrection);
+    return {
+      result,
+      prevCenter,
+      nextCenter,
+      offset,
+    };
+  }
+
+  protected commitLayout(
+    context: StrictLayoutContext,
+    layoutResult: Required<LayoutResult>
+  ) {
+    const { target } = context;
+    const {
+      result: { width, height },
+      nextCenter,
+    } = layoutResult;
+    // set dimensions
+    target.set({ width, height });
+    // layout descendants
+    this.layoutObjects(context, layoutResult);
+    //  set position
+    // in `initialization` we do not account for target's transformation matrix
+    if (context.type === 'initialization') {
+      // TODO: what about strokeWidth?
+      const origin = nextCenter.add(
+        new Point(width, height).multiply(
+          new Point(
+            resolveOrigin(target.originX),
+            resolveOrigin(target.originY)
+          )
+        )
+      );
+      target.set({ left: context.x ?? origin.x, top: context.y ?? origin.y });
+    } else {
+      target.setPositionByOrigin(nextCenter, CENTER, CENTER);
+      // invalidate
+      target.setCoords();
+      target.set({ dirty: true });
+    }
+  }
+
+  protected layoutObjects(
+    context: StrictLayoutContext,
+    layoutResult: Required<LayoutResult>
+  ) {
+    const { target } = context;
+    //  adjust objects to account for new center
+    (context.type !== 'initialization' || !context.objectsRelativeToGroup) &&
+      target.forEachObject((object) => {
+        object.group === target &&
+          this.layoutObject(context, layoutResult, object);
+      });
+    // adjust clip path to account for new center
+    context.strategy.shouldLayoutClipPath(context) &&
+      this.layoutObject(context, layoutResult, target.clipPath as FabricObject);
+  }
+
+  /**
+   * @param {FabricObject} object
+   * @param {Point} offset
+   */
+  protected layoutObject(
+    context: StrictLayoutContext,
+    { offset }: Required<LayoutResult>,
+    object: FabricObject
+  ) {
+    object.setRelativeXY(object.getRelativeXY().add(offset));
```

Comment:
please this is exactly:
```suggestion
    object.left += offset.x;
    object.top += offset.y
```
let's not make it more complicated than it is until is really needed

Reply:
true, but I want to move away from left/top

Reply:
let's be clear.
Future plans not discussed do not count.
As of today we have left and top.
If we had x and y this suggestion would have been:

```js
    object.x += offset.x;
    object.y += offset.y
```

Reply:
for now I will live with it